File: contribuyenteApi/ContribuyenteApi.Domain/Entities/Contribuyente.cs

namespace ContribuyenteApi.Domain.Entities;

public class Contribuyente
{
    public string RncCedula { get; set; } = string.Empty;
    public string Nombre { get; set; } = string.Empty;
    public string Tipo { get; set; } = string.Empty;
    public string Estatus { get; set; } = string.Empty;
    
    // Relación 1 a N
    public ICollection<ComprobanteFiscal> Comprobantes { get; set; } = new List<ComprobanteFiscal>();
}

File: contribuyente-frontend/src/types/contribuyente.ts
export interface Contribuyente {
    rncCedula: string;
    nombre: string;
    tipo: string;
    estatus: string;
}

export interface Comprobante {
    ncf: string;
    monto: number;
    itbis18: number;
}

export interface ContribuyenteDetalle {
    rncCedula: string;
    comprobantes: Comprobante[];
    totalItbis: number;
}
